pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -282,6 +284,15 @@ func (ssr *Snapshotter) takeFullSnapshot(isFinal bool) (*brtypes.Snapshot, error
 	// close previous watch and client.
 	ssr.closeEtcdClient()
 
+	var err error
+
+	// Update the snapstore object before taking the full snapshot
+	// Refer: https://github.com/gardener/etcd-backup-restore/issues/422
+	ssr.store, err = snapstore.GetSnapstore(ssr.snapstoreConfig)
```

Comment:
Creating a new store object is related to some overhead, not only re-reading the credentials but also instantiating a new client, connection etc. (depends on cloud provider). WDYT about creating and saving a hash value of the currently used credentials and only create an entirely new Snapstore if the hash has changed?